### PR TITLE
feat: specify a custom "terms of service" link

### PR DESCRIPTION
**Pull request by aslilac**
Adds support for requiring users to accept a "terms of service" when logging in

Screenshot 2024-04-25 at 12 54 03 PM

### CODE REVIEW

File: coderd/userauth.go [hunk 1]
```diff
@@ -486,7 +487,7 @@ func (api *API) userAuthMethods(rw http.ResponseWriter, r *http.Request) {
 
 // @Summary OAuth 2.0 GitHub Callback
 // @ID oauth-20-github-callback
-// @Security CoderSessionToken
+// @Security CoderSessionTokens
```
Comment (aslilac) [inline coderd/userauth.go:490]: some formatter/linter did this

File: site/src/api/typesGenerated.ts [hunk 2]
```diff
@@ -445,6 +446,7 @@ export interface DeploymentValues {
   readonly allow_workspace_renames?: boolean;
   readonly healthcheck?: HealthcheckConfig;
   readonly cli_upgrade_message?: string;
+  readonly terms_of_service_link?: string;
```
Comment (BrunoQuaresma) [inline site/src/api/typesGenerated.ts:449]: Why is this here and also in the `AuthMethods` type?

File: site/src/pages/LoginPage/SignInForm.tsx [hunk 3]
```diff
@@ -83,6 +85,10 @@ export const SignInForm: FC<SignInFormProps> = ({
   const passwordEnabled = authMethods?.password.enabled ?? true;
   const applicationName = getApplicationName();
 
+  const [tosAccepted, setTosAccepted] = useState(false);
```
Comment (BrunoQuaresma) [inline site/src/pages/LoginPage/SignInForm.tsx:88]: I was expecting this to be filled by some query data.

Comment (aslilac) [reply site/src/pages/LoginPage/SignInForm.tsx:88]: we're adding this for a single customer, and Kyle doesn't want any more backend than is absolutely necessary.

File: site/src/api/typesGenerated.ts (hunk 2 above)
Comment (aslilac) [reply site/src/api/typesGenerated.ts:449]: - it has to be here, because this is the type that all of our config from env variables and args gets loaded into
- we want to keep /users/authmethods as the only query that the login screen needs to make, so the custom terms of service gets served from there

**Review by BrunoQuaresma — APPROVED** [review]

**Comment by aslilac** [comment]
https://github.com/coder/coder/assets/418348/942b7c89-8e6c-49fa-8048-c116925ee0a1
